lib/src/native/realm_core.dart

```diff
@@ -125,7 +125,7 @@ class _RealmCore {
         classInfo.num_properties = schemaObject.properties.length;
         classInfo.num_computed_properties = 0;
         classInfo.key = RLM_INVALID_CLASS_KEY;
-        classInfo.flags = realm_class_flags.RLM_CLASS_NORMAL;
+        classInfo.flags = schemaObject.baseType.toFlags();
```

Comment:
The reason why I don't like it is because we're tying a public API (`ObjectType`) to an internal one - especially since Core has been reluctant to commit to fixed values for their enums. But I agree, if we can rely on Core enums not changing values, I'd be fine to hardcode the values.

Reply:
I changed it to store both the name and the flags.